### PR TITLE
Change `IAnsiConsole` to render `IRenderable` instead of `IEnumerable<Segment>`

### PR DESCRIPTION
**Pull request by patriksvensson**
This makes it possible for encoders to output better representation of the actual objects instead of working with chopped up segments.

* `IAnsiConsole.Write` now takes an IRenderable instead of segments
* Calculating cell width does no longer require a render context
* Removed `RenderContext.LegacyConsole`
* Removed `RenderContext.Encoding`
* Added `Capabilities.Unicode`

### CODE REVIEW

File: src/Spectre.Console/Internal/Backends/Ansi/AnsiConsoleCursor.cs [hunk 1]
```diff
@@ -16,11 +15,11 @@ public void Show(bool show)
         {
             if (show)
             {
-                _backend.Render(new[] { Segment.Control("\u001b[?25h") });
+                _backend.Write(new ControlSequence("\u001b[?25h"));
```
Comment (phil-scott-78) [inline src/Spectre.Console/Internal/Backends/Ansi/AnsiConsoleCursor.cs:18]: a thought for the future but not this PR - be nice if `ControlSequence` had some of these values as constants, or in the case of the ones like "move up x characters" there is a function that generates the value. To the untrained eye `\u001b[?25h` not only looks like gibberish, but is pretty difficult to google.

**Review by phil-scott-78 — APPROVED** [review]

**Comment by phil-scott-78** [comment]
Finally got a chance to sit down and play with the update last night. No surprises on my end, and the code looks good to me.

But with it being such a big change I wanted to make sure we both turned the key at the same on this launch before I merge it so just let me know when you are all set.

File: src/Spectre.Console/Internal/Backends/Ansi/AnsiConsoleCursor.cs (hunk 1 above)
Comment (patriksvensson) [reply src/Spectre.Console/Internal/Backends/Ansi/AnsiConsoleCursor.cs:18]: Absolutely! I will send another PR for this!

**Comment by patriksvensson** [comment]
@phil-scott-78 Actually noticed a bug so might be a good idea to wait merging this for the time being

**Pull request merged**